controllers/ocimachine_controller.go

```diff
@@ -189,6 +195,10 @@ func (r *OCIMachineReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Ma
 			&infrastructurev1beta2.OCICluster{},
 			handler.EnqueueRequestsFromMapFunc(r.OCIClusterToOCIMachines()),
 		).
+		Watches(
```

Comment:
While testing the self managed nodes today, I noticed that there is a timing issue. The OKE cluster creation is taking slightly longer than expected, and since we are not watching OCIManagedCluster, the machine reconciliation is not happening quickly as it is in exponential backoff loop(data secret lookup error as the control plane is not initialized).I had noticed this behaviour earlier, but this is more noticeable today.